### PR TITLE
Shir/feature/socket io notifications

### PR DESCRIPTION
**Pull request by shirbalas**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+const port = process.env.PORT || 3001;
+app.set("port", port);
+// app.listen(port, () => {
+//   console.log("Server running at port " + port);
+// });
+
+
+//-------------- socket io notification -----------------//
+
```
Comment (barakeitan) [inline app.js:34]: what is this huge commented code ? @shirbalas

File: combined.log [hunk 2]
```diff
@@ -0,0 +1,39 @@
+2022-10-29 14:38:38.958 INFO: 	user 1234 brodcast message: Hello -ctXvTvKiCSBNKaZAAAB :)
```
Comment (barakeitan) [inline combined.log:1]: consider adding combined and error logs to git ignore @shirbalas

File: error.log [hunk 3]
```diff
@@ -0,0 +1 @@
+2022-10-29 14:38:38.960 ERROR: 	user 1234 brodcast message: Hello -ctXvTvKiCSBNKaZAAAB :)
```
Comment (barakeitan) [inline error.log:1]: same as combined log comment @shirbalas

File: package.json [hunk 4]
```diff
@@ -28,4 +28,4 @@
     "mongoose": "^6.7.0",
     "mongoose-slug-updater": "^3.3.0"
   }
-}
+}
```
Comment (barakeitan) [inline package.json:31]: Redundant, remove the spacing @shirbalas

File: public/js/logger.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (barakeitan) [inline public/js/logger.js:1]: you have the same code and the same exports in app.js
if you make a file out of it and not use the logging abilities in the app.js then remove the double code @shirbalas

File: public/js/wsClient.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+const { io } = require("socket.io-client");
+const {logger} = require('./logger.js');
+const port = process.env.PORT || 3001;
+const socket = io(`ws://localhost:${port}`);
```
Comment (barakeitan) [inline public/js/wsClient.js:4]: consider to change the address of the socket io to the dotenv and choose an address less local @shirbalas

File: public/js/wsServer.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+function getByValue(searchValue) {
+    for (let [key, value] of users.entries()) {
+      if (value === searchValue)
+        return key;
+    }
+  }
+
+http.listen(port, () => console.log(`ws listening on http://localhost:${port}`) );
```
Comment (barakeitan) [inline public/js/wsServer.js:42]: same note about the socket address @shirbalas

**Review by barakeitan — COMMENTED** [review]
good work on documenting the code, only minor changes

**Comment by shirbalas** [comment]
need to be retested

**Pull request closed without merging**